### PR TITLE
Implementing Doe's new banners and signs.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
Random debug types to show the concepts:
image
image

Icons by @PlayerDeer.

### CODE REVIEW

File: code/game/objects/structures/banners.dm [hunk 1]
```diff
@@ -64,95 +79,279 @@
...
-		var/image/I = image(banner.icon, "banner_base")
+	if(!istype(banner))
+		return
+
+	var/image/I = image(banner.icon, "[banner.icon_state]-hanging")
+	I.appearance_flags |= RESET_COLOR
+	I.color = banner.color
+	add_overlay(I)
+
+	if(banner.trim_color)
```
Comment (out-of-phaze) [inline code/game/objects/structures/banners.dm]: kind of feel like the decals should be added to the banner state first, with `BLEND_INSET_OVERLAY`, and then we apply the trim on top to keep the layering sane. that way you could have user-added decals like stripes/crosses/etc that aren't mutually exclusive with other decals and work regardless of banner shape. this probably works fine as-is i just wanted to point out that inset overlay blend mode could be useful here now or in the future

**Review by out-of-phaze — DISMISSED** [review]
presumably at some point modpacks should be able to add additional banner icons, e.g. corporate logos as decals in the corporate modpack, holy symbols as decals in the mundane/modern earth modpack, etc.

would also be nice to, in the future, have it list decals on examine, including their color if supported. `"This is a <font color="white">banner</font> with a <font color="red">sword</span> embroidered on it."`, `"This is a walnut sign with a beer mug embroidered on it." etc.

for now this looks good, might be worth adding a unit test in the future that checks if the various states exist for every banner type's base_icon_state, like banner-hanging etc.

File: code/game/objects/structures/banners.dm (hunk 1 above)
Comment (MistakeNot4892) [reply code/game/objects/structures/banners.dm]: Done.

**Comment by MistakeNot4892** [comment]
````
presumably at some point modpacks should be able to add additional banner icons, e.g. corporate logos as decals in the corporate modpack, holy symbols as decals in the mundane/modern earth modpack, etc.
````
Symbols are now decls, so this is done.

````
would also be nice to, in the future, have it list decals on examine, including their color if supported. "This is a <font color="white">banner</font> with a <font color="red">sword</span> embroidered on it.", `"This is a walnut sign with a beer mug embroidered on it." etc.
````
Simple version of this is done.

**Review by out-of-phaze — APPROVED** [review]

**Pull request merged**